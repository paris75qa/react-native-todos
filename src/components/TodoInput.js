import React, { Component, PropTypes } from 'react';
import { TouchableHighlight, View, TextInput, Text, StyleSheet } from 'react-native';


const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
  },
  input: {
    flex: 1,
    borderColor: 'gray',
    borderWidth: 1,
    height: 40,
    paddingLeft: 5,
  },
  button: {
    flex: 0,
    marginLeft: 5,
    marginRight: 5,
  },
});
class TodoInput extends Component {
  constructor(props) {
    super(props);
    this.onInputChange = this.onInputChange.bind(this);
    this.onClickButton = this.onClickButton.bind(this);
    this.state = {
      inputValue: "",
    }
  }

  onInputChange(text){
    this.setState({
      inputValue: text,
    });
  }

  onClickButton(){
    this.setState({
      inputValue: "",
    });
    this.props.onAddTodo(this.state.inputValue);
  }

  render() {
    return (
        <View style={styles.container}>
          <TextInput
              style={styles.input}
              value={this.state.inputValue}
              onChangeText={this.onInputChange}
          />
          <TouchableHighlight
              style={styles.button}
              onPress={this.onClickButton}>
              <Text>Ajouter</Text>
          </TouchableHighlight>
        </View>
    );
  }
}

TodoInput.propTypes = {
  onAddTodo: PropTypes.func.isRequired,
}


export default TodoInput;
