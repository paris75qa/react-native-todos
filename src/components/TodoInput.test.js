import React from 'react';
import TodoInput from './TodoInput';
import renderer from 'react-test-renderer';

it('TodoInput renders correctly', () => {
  const tree = renderer.create(
    <TodoInput onAddTodo={()=>{}}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
