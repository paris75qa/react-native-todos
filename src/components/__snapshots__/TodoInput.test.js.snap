// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "flex": 1,
        "height": 40,
        "paddingLeft": 5,
      }
    }
    value=""
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 0,
          "marginLeft": 5,
          "marginRight": 5,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Ajouter
    </Text>
  </View>
</View>
`;
